packages/core/src/shared/sam/cli/samCliLocator.ts

```diff
@@ -31,11 +36,11 @@ export class SamCliLocationProvider {
         const cachedLoc = forceSearch ? undefined : SamCliLocationProvider.cachedSamLocation
 
         // Avoid searching the system for `sam` (especially slow on Windows).
-        if (cachedLoc && (await SamCliLocationProvider.isValidSamLocation(cachedLoc.path))) {
+        if (cachedLoc && (cachedLoc.verified || (await SamCliLocationProvider.isValidSamLocation(cachedLoc.path)))) {
+            cachedLoc.verified = true
```

Comment:
Seems like `cachedLoc.verified` should be a property of `SamCliLocationProvider`